services/cd-service/pkg/repository/transformer.go

```diff
@@ -596,15 +596,17 @@ func writeCommitData(ctx context.Context, sourceCommitId string, sourceMessage s
 	return nil
 }
 
-func writeNextPrevInfo(sourceCommitId string, otherCommitId string, fieldSource string, fs billy.Filesystem) error {
+func writeNextPrevInfo(ctx context.Context, sourceCommitId string, otherCommitId string, fieldSource string, application string, fs billy.Filesystem) error {
 
 	otherCommitId = strings.ToLower(otherCommitId)
 	sourceCommitDir := commitDirectory(fs, sourceCommitId)
 
 	otherCommitDir := commitDirectory(fs, otherCommitId)
 
 	if _, err := fs.Stat(otherCommitDir); err != nil {
-		return err
+		logger.FromContext(ctx).Sugar().Warnf(
+			"%s application does not contain any commit with the provided id (%s). Ignoring provided information.", application, otherCommitId)
+		return nil
```

Comment:
The app is not part of the `othercommitDir` path. So I don't see why it's relevant, apart from contextual information in the warning.
I'd also like to see the directory name printed.

```suggestion
		logger.FromContext(ctx).Sugar().Warnf(
			"Could not find the previous commit while trying to create a new release for commit %s and application %s. This is expected when `git.enableWritingCommitData` was just turned on, however it should not happen multiple times.", otherCommitId, application, otherCommitDir)
		return nil
```